aten/src/ATen/mps/MPSAllocator.mm

```diff
@@ -202,6 +212,9 @@
     block_found =
         // Attempt allocate
         alloc_buffer(params) ||
+        // Callbacks might release more memory (eg. by forcing a GC in the host language) thus
+        // we can retry getting a free buffer in the pool, before trying to alloc again.
+        (trigger_callbacks() && get_free_buffer(params)) ||
```

Comment:
I think we should separate out the change of `recursive_mutex` with the Release event type.

Reply:
I changed them to be in 2 different commits.